### ISSUE TITLE
Support Spring Data Kay

### ISSUE THREAD

**Issue by wilkinsona** [spring-projects/spring-session#677]
The errors can be reproduced using 

```
$ ./gradlew clean springIoCheck -PplatformVersion=Cairo-BUILD-SNAPSHOT
```

Spring Data Kay is the Spring Data release train that requires Spring Framework 5 and Java 8. It will be part of Spring Boot 2.0 and Spring IO Platform Cairo. It can be seen as Spring Data 2.0 in that breaking changes to user-facing APIs are to be expected.

We've seen some evidence of these breaking changes already. For example, `AbstractMongoSessionConverter` has been broken by them:

```
Caused by: java.lang.NoSuchMethodError: org.springframework.data.mongodb.core.IndexOperations.ensureIndex(Lorg/springframework/data/mongodb/core/index/IndexDefinition;)V
	at org.springframework.session.data.mongo.AbstractMongoSessionConverter.ensureIndexes(AbstractMongoSessionConverter.java:79)
	at org.springframework.session.data.mongo.MongoOperationsSessionRepository.ensureIndexesAreCreated(MongoOperationsSessionRepository.java:123)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor$LifecycleElement.invoke(InitDestroyAnnotationBeanPostProcessor.java:366)
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor$LifecycleMetadata.invokeInitMethods(InitDestroyAnnotationBeanPostProcessor.java:311)
	at org.springframework.beans.factory.annotation.InitDestroyAnnotationBeanPostProcessor.postProcessBeforeInitialization(InitDestroyAnnotationBeanPostProcessor.java:134)
	... 39 more
```

We need to figure out how to support Spring Data Kay in Spring Session.

**Comment by rwinch** [comment]
@wilkinsona It appears there are lots of missing dependency mappings when testing against Cario. Right now on spring-session master I get:

```
$ gradlew springioCheck --stacktrace --refresh-dependencies --no-daemon -PplatformVersion=Cario-BUILD-SNAPSHOT
...
:samples:websocket:clean UP-TO-DATE
:spring-session:springIoAlternativeDependenciesCheck
:spring-session:springIoDependencyVersionMappingCheck FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':spring-session:springIoDependencyVersionMappingCheck'.
> The following direct dependencies do not have Spring IO versions: 
      - org.springframework.data:spring-data-redis
      - com.hazelcast:hazelcast
      - org.springframework.data:spring-data-gemfire
      - org.springframework:spring-jdbc
      - org.springframework.data:spring-data-mongodb
      - org.springframework:spring-context
      - org.springframework:spring-web
      - org.springframework:spring-messaging
      - org.springframework:spring-websocket
      - org.springframework.security:spring-security-core
      - org.springframework.security:spring-security-web
      - javax.servlet:javax.servlet-api
      - org.springframework:spring-context
      - org.springframework:spring-context
  Please refer to the plugin's README for further instructions: https://github.com/spring-projects/gradle-plugins/tree/master/spring-io-plugin#dealing-with-unmapped-dependencies

* Try:
Run with --info or --debug option to get more log output.

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':spring-session:springIoDependencyVersionMappingCheck'.
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:69)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.execute(ExecuteActionsTaskExecuter.java:46)
        at org.gradle.api.internal.tasks.execution.PostExecutionAnalysisTaskExecuter.execute(PostExecutionAnalysisTaskExecuter.java:35)
        at org.gradle.api.internal.tasks.execution.SkipUpToDateTaskExecuter.execute(SkipUpToDateTaskExecuter.java:64)
        at org.gradle.api.internal.tasks.execution.ValidatingTaskExecuter.execute(ValidatingTaskExecuter.java:58)
        at org.gradle.api.internal.tasks.execution.SkipEmptySourceFilesTaskExecuter.execute(SkipEmptySourceFilesTaskExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipTaskWithNoActionsExecuter.execute(SkipTaskWithNoActionsExecuter.java:52)
        at org.gradle.api.internal.tasks.execution.SkipOnlyIfTaskExecuter.execute(SkipOnlyIfTaskExecuter.java:53)
        at org.gradle.api.internal.tasks.execution.ExecuteAtMostOnceTaskExecuter.execute(ExecuteAtMostOnceTaskExecuter.java:43)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:233)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter$EventFiringTaskWorker.execute(DefaultTaskGraphExecuter.java:215)
        at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.processTask(AbstractTaskPlanExecutor.java:74)
        at org.gradle.execution.taskgraph.AbstractTaskPlanExecutor$TaskExecutorWorker.run(AbstractTaskPlanExecutor.java:55)
        at org.gradle.execution.taskgraph.DefaultTaskPlanExecutor.process(DefaultTaskPlanExecutor.java:32)
        at org.gradle.execution.taskgraph.DefaultTaskGraphExecuter.execute(DefaultTaskGraphExecuter.java:113)
        at org.gradle.execution.SelectedTaskExecutionAction.execute(SelectedTaskExecutionAction.java:37)
        at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:37)
        at org.gradle.execution.DefaultBuildExecuter.access$000(DefaultBuildExecuter.java:23)
        at org.gradle.execution.DefaultBuildExecuter$1.proceed(DefaultBuildExecuter.java:43)
        at org.gradle.execution.DryRunBuildExecutionAction.execute(DryRunBuildExecutionAction.java:32)
        at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:37)
        at org.gradle.execution.DefaultBuildExecuter.execute(DefaultBuildExecuter.java:30)
        at org.gradle.initialization.DefaultGradleLauncher$4.run(DefaultGradleLauncher.java:186)
        at org.gradle.internal.Factories$1.create(Factories.java:22)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:91)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:53)
        at org.gradle.initialization.DefaultGradleLauncher.doBuildStages(DefaultGradleLauncher.java:183)
        at org.gradle.initialization.DefaultGradleLauncher.access$200(DefaultGradleLauncher.java:33)
        at org.gradle.initialization.DefaultGradleLauncher$1.create(DefaultGradleLauncher.java:112)
        at org.gradle.initialization.DefaultGradleLauncher$1.create(DefaultGradleLauncher.java:106)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:91)
        at org.gradle.internal.progress.DefaultBuildOperationExecutor.run(DefaultBuildOperationExecutor.java:63)
        at org.gradle.initialization.DefaultGradleLauncher.doBuild(DefaultGradleLauncher.java:106)
        at org.gradle.initialization.DefaultGradleLauncher.run(DefaultGradleLauncher.java:92)
        at org.gradle.launcher.exec.GradleBuildController.run(GradleBuildController.java:66)
        at org.gradle.tooling.internal.provider.ExecuteBuildActionRunner.run(ExecuteBuildActionRunner.java:28)
        at org.gradle.launcher.exec.ChainingBuildActionRunner.run(ChainingBuildActionRunner.java:35)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:41)
        at org.gradle.launcher.exec.InProcessBuildActionExecuter.execute(InProcessBuildActionExecuter.java:26)
        at org.gradle.tooling.internal.provider.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:79)
        at org.gradle.tooling.internal.provider.ContinuousBuildActionExecuter.execute(ContinuousBuildActionExecuter.java:51)
        at org.gradle.launcher.cli.RunBuildAction.run(RunBuildAction.java:51)
        at org.gradle.internal.Actions$RunnableActionAdapter.execute(Actions.java:173)
        at org.gradle.launcher.cli.CommandLineActionFactory$ParseAndBuildAction.execute(CommandLineActionFactory.java:244)
        at org.gradle.launcher.cli.CommandLineActionFactory$ParseAndBuildAction.execute(CommandLineActionFactory.java:217)
        at org.gradle.launcher.cli.JavaRuntimeValidationAction.execute(JavaRuntimeValidationAction.java:33)
        at org.gradle.launcher.cli.JavaRuntimeValidationAction.execute(JavaRuntimeValidationAction.java:24)
        at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:33)
        at org.gradle.launcher.cli.ExceptionReportingAction.execute(ExceptionReportingAction.java:22)
        at org.gradle.launcher.cli.CommandLineActionFactory$WithLogging.execute(CommandLineActionFactory.java:210)
        at org.gradle.launcher.cli.CommandLineActionFactory$WithLogging.execute(CommandLineActionFactory.java:174)
        at org.gradle.launcher.Main.doAction(Main.java:33)
        at org.gradle.launcher.bootstrap.EntryPoint.run(EntryPoint.java:45)
        at org.gradle.launcher.bootstrap.ProcessBootstrap.runNoExit(ProcessBootstrap.java:60)
        at org.gradle.launcher.bootstrap.ProcessBootstrap.run(ProcessBootstrap.java:37)
        at org.gradle.launcher.GradleMain.main(GradleMain.java:23)
        at org.gradle.wrapper.BootstrapMainStarter.start(BootstrapMainStarter.java:31)
        at org.gradle.wrapper.WrapperExecutor.execute(WrapperExecutor.java:108)
        at org.gradle.wrapper.GradleWrapperMain.main(GradleWrapperMain.java:61)
Caused by: org.gradle.api.InvalidUserDataException: The following direct dependencies do not have Spring IO versions: 
    - org.springframework.data:spring-data-redis
    - com.hazelcast:hazelcast
    - org.springframework.data:spring-data-gemfire
    - org.springframework:spring-jdbc
    - org.springframework.data:spring-data-mongodb
    - org.springframework:spring-context
    - org.springframework:spring-web
    - org.springframework:spring-messaging
    - org.springframework:spring-websocket
    - org.springframework.security:spring-security-core
    - org.springframework.security:spring-security-web
    - javax.servlet:javax.servlet-api
    - org.springframework:spring-context
    - org.springframework:spring-context
Please refer to the plugin's README for further instructions: https://github.com/spring-projects/gradle-plugins/tree/master/spring-io-plugin#dealing-with-unmapped-dependencies
        at io.spring.gradle.springio.CheckPlatformDependenciesBeforeResolveAction$_execute_closure1.doCall(CheckPlatformDependenciesBeforeResolveAction.groovy:42)
        at org.gradle.listener.ClosureBackedMethodInvocationDispatch.dispatch(ClosureBackedMethodInvocationDispatch.java:40)
        at org.gradle.listener.ClosureBackedMethodInvocationDispatch.dispatch(ClosureBackedMethodInvocationDispatch.java:25)
        at org.gradle.internal.event.AbstractBroadcastDispatch.dispatch(AbstractBroadcastDispatch.java:44)
        at org.gradle.internal.event.BroadcastDispatch.dispatch(BroadcastDispatch.java:79)
        at org.gradle.internal.event.BroadcastDispatch.dispatch(BroadcastDispatch.java:30)
        at org.gradle.internal.dispatch.ProxyDispatchAdapter$DispatchingInvocationHandler.invoke(ProxyDispatchAdapter.java:93)
        at com.sun.proxy.$Proxy23.afterResolve(Unknown Source)
        at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.resolveGraphIfRequired(DefaultConfiguration.java:406)
        at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.resolveNow(DefaultConfiguration.java:371)
        at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration.getResolvedConfiguration(DefaultConfiguration.java:364)
        at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration_Decorated.getResolvedConfiguration(Unknown Source)
        at org.gradle.internal.metaobject.BeanDynamicObject$MetaClassAdapter.getProperty(BeanDynamicObject.java:197)
        at org.gradle.internal.metaobject.BeanDynamicObject.getProperty(BeanDynamicObject.java:150)
        at org.gradle.internal.metaobject.CompositeDynamicObject.getProperty(CompositeDynamicObject.java:55)
        at org.gradle.internal.metaobject.AbstractDynamicObject.getProperty(AbstractDynamicObject.java:60)
        at org.gradle.api.internal.artifacts.configurations.DefaultConfiguration_Decorated.getProperty(Unknown Source)
        at io.spring.gradle.springio.DependencyVersionMappingCheckTask.checkVersionMapping(DependencyVersionMappingCheckTask.groovy:38)
        at org.gradle.internal.reflect.JavaMethod.invoke(JavaMethod.java:75)
        at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.doExecute(DefaultTaskClassInfoStore.java:136)
        at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.execute(DefaultTaskClassInfoStore.java:129)
        at org.gradle.api.internal.project.taskfactory.DefaultTaskClassInfoStore$StandardTaskAction.execute(DefaultTaskClassInfoStore.java:118)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:623)
        at org.gradle.api.internal.AbstractTask$TaskActionWrapper.execute(AbstractTask.java:606)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeAction(ExecuteActionsTaskExecuter.java:80)
        at org.gradle.api.internal.tasks.execution.ExecuteActionsTaskExecuter.executeActions(ExecuteActionsTaskExecuter.java:61)
        ... 58 more


BUILD FAILED

Total time: 54.432 secs
```

Any ideas why this is happening?

**Comment by wilkinsona** [comment]
@rwinch That's very strange. The latest master (a7b5f86bcd5570ccbaaf5d6a95d065bdd8e5071f) works for me in that `springIoDependencyVersionMappingCheck` passes and I then see test failures due to breaking API changes.

**Comment by rwinch** [comment]
@wilkinsona I still get failures around `springIoDependencyVersionMappingCheck`. Do you run the command I provided in the root folder?

**Comment by wilkinsona** [comment]
@rwinch Looks like a typo in the Platform's version. You need `Cairo-BUILD-SNAPSHOT` rather than `Cario-BUILD-SNAPSHOT`.

**Comment by jxblum** [comment]
@rwinch, @wilkinsona - never mind; argh!

It's not `spring-boot-starter-redis`; it is `spring-boot-starter-data-redis`.

**Comment by jxblum** [comment]
@rwinch - I am able to compile now, but I am not able to `clean build install` (complete with tests).

Based on earlier comments (above), I suspect the test failures are expected based on the API breaking changes.  I'd like to get the build working before moving forward on upgrading _Spring Session_ to _Spring Data Kay_.

Additionally, since _Spring IO Platform_ **Cairo** and _Spring Boot_ **2.0** will be based on _Spring_ **5** and **Java 8**, I also gather that we need to bump the _Spring Boot_ version from 1.5 to 2.0 in _Spring Session_ **2.0**.

I initially tried to build and test `master` with just _Spring Boot_ **1.5**, but that is not working, even after updating the test annotations based on the API changes from Boot 1.4 to 1.5.  No doubt I have a configuration problem, but not sure what that is yet.

**Comment by vpavic** [comment]
IMO we can't upgrade to Boot 2.0 since Boot itself provides dependency management for Spring Session.

Current `master` (i.e. 2.0) is natural target for Boot 2.0 however Session 2.0 needs to be GA before Boot 2.0 does in order to get included. If we upgraded to Boot 2.0 it would mean that Session 2.0 at the time it reaches GA would depend on an unstable release of Boot 2.0.

**Comment by jxblum** [comment]
@vpavic - yes, excellent point.  After all the discussions between @rwinch and @wilkinsona, I was only thinking about the _Spring IO Platform_ (which builds on _Spring Boot_, actually; parent pom), but clearly _Spring Boot_ supports and has several integration points (dependencies) on _Spring Session_.

However, I will say, we definitely need to make the sure _Spring Session_ dependencies are better kept up-to-date, and more succinctly, kept in-sync with _Spring Boot's_/Spring IO Platform's harmonized set of dependency versions.  There are (were now) several instances where _Spring Session_ has significantly fallen behind.  Not, even the _Spring Data_ module dependencies were in-sync with each other.  I am trying to incorporate the _Spring Data_ BOM for this very reason, somehow; Unfortunately I am not quite as adept of a Gradle user as I am with Maven, yet.

Anyway... look for some updates soon; I am nearly there.

**Comment by jxblum** [comment]
This Issue has been completed and checked into to `master`.

**Comment by vpavic** [comment]
@jxblum It appears you've pushed some commits related to #697 together with your fix for this issue. Did you intend to do that?

**Comment by rwinch** [comment]
@vpavic Thanks for helping to keep an eye on things. The changes for #697 were somewhat necessary as Spring Boot 1.5 was not working with Spock tests. We could have probably found a way around this, but given we wanted to move to a completely Java stack anyways, we thought it was best to just include those in the fixes.

**Comment by vpavic** [comment]
@rwinch I see, thanks for clearing that up. I saw no mention of changes on #697 itself so that made me suspect the commits related to it were pushed by accident.

**Comment by rwinch** [comment]
I think the problem is that the commit messages were not cleaned up. This is unfortunate, but it is done. We still have to merge those changes into 1.4.x as well. We can cleanup the commit messages then

**Comment by jxblum** [comment]
@rwinch, @vpavic - My apologies; my mistake.  While I have resolved this issue, the commit contained 18 separate commits (I did not squash them; well, only a few related commits).  So, in essence, all the commits representing the other individual issues are there and those had to be fixed before I could move forward with the _Spring Data_ modules upgrade.  I should have looked for the corresponding issues before committing to master.  However, I have noticed the commit history is not always necessarily tied to a GH Issue, although, I generally feel that it is good practice to do so.

**Comment by jxblum** [comment]
@rwinch - I am noting a few details of this upgrade I want to bring into focus before closing out this Issue.

1.  First, _Spring Session_ has been stuck on _Spring Data Redis_ **1.7.1.RELEASE** for quite some time due to an API [incompatibility between SD Redis](https://github.com/spring-projects/spring-data-redis/blob/1.7.2.RELEASE/src/main/java/org/springframework/data/redis/serializer/JdkSerializationRedisSerializer.java#L53) (introduced in SDR 1.7.2.RELEASE, thus causing the incompatibility) and the core _Spring Framework_, particularly after _Spring_ 3 (technically [introduced in Spring 4.2.1.RELEASE](https://github.com/spring-projects/spring-framework/blob/v4.2.1.RELEASE/spring-core/src/main/java/org/springframework/core/serializer/support/DeserializingConverter.java#L50-L58)), while the other _Spring Data_ modules (e.g. SD GemFire and SD MongoDB) have progressed forward.  

This has caused SD Redis to get out-of-sync with other primary _Spring Data_ modules (e.g. SD GemFire and SD MongoDB).

My solution to sync problem is to enforce primary SD module parity by using the Spring Data BOM file.

We can either due this by importing the SD BOM (which is how it currently is in this fix), or indirectly through Spring Boot, since it defines the `[spring-data-releasetrain.version](https://github.com/spring-projects/spring-boot/blob/v1.5.0.RC1/spring-boot-dependencies/pom.xml#L156)` build property.  I am leaning towards the later.

Furthermore, it upgrading SD Redis causes the `**spring3Test**` check (currently, disabled) to fail.  Specifically, only the _Spring Session Data Redis_ related tests are failing, and in fact, all other SD module spring3Test checks pass!

Since _Spring_ 3 is technically still supported (though there are no longer any other planned service release in the 3.x line), this could be resolved if this [constructor](https://github.com/spring-projects/spring-framework/blob/v4.2.1.RELEASE/spring-core/src/main/java/org/springframework/core/serializer/support/DeserializingConverter.java#L50-L58) was back ported to `spring-core` **3.2.x** (i.e. 3.2.19.RELEASE) thereby allowing the spring3Test check to persist for as long as _Spring_ **3.2.x** is listed on the _Spring Framework_ [projects page](http://projects.spring.io/spring-framework/).

2. Second, while I have [imported](https://github.com/spring-projects/spring-session/blob/master/spring-session/build.gradle#L56-L58) the _Spring Data_ BOM file, something is running interference and preventing the individual `spring-session-data-<modules>` from being set with the correct _Spring Data_ module version based on the SD BOM file, which required me to explicitly still specify the SD module versions, [for instance](https://github.com/spring-projects/spring-session/blob/master/spring-session-data-redis/build.gradle#L11).

This most likely caused by my Gradle ignorance, but I was (half) expecting SD BOM to be successfully applied given the `:spring-session` module is [declared in the dependencies](https://github.com/spring-projects/spring-session/blob/master/spring-session-data-redis/build.gradle#L10) of the individual spring-`session-data-<module>`.

Even if I explicitly imported the SD BOM file in the individual `spring-session-data-<module>` build.gradle files, like so...

```xml
dependencyManagement {
    imports {
        mavenBom "org.springframework.data:spring-data-releasetrain:$springDataReleaseTrainVersion"
    }   
    springIoTestRuntime {
        imports {
            mavenBom "io.spring.platform:platform-bom:${springIoVersion}"
        }
    }
}
```
It still does not work without a version.  Otherwise, the corresponding SD module has NO version (or would acquire the wrong version, with tweaks) and results in compilation and/or test errors during the build.

I can only conclude this had something to do with the _Spring IO Platform_ BOM file [import](https://github.com/spring-projects/spring-session/blob/master/spring-session-data-redis/build.gradle#L21-L23), above and beyond even the explicit SD BOM import (as shown above).

Of course, the _Spring IO Platform_ BOM [inherits](https://github.com/spring-io/platform/blob/master/platform-bom/pom.xml#L6-L11) from _spring-boot-starter-parent_ (and indirectly, the _spring-boot-dependencies_ POM), and since _Spring Boot_ was set to **1.5.0.RC1** in my commit, this is the LCD, and thus will [inherit the Ingalls SD Release Trains](https://github.com/spring-projects/spring-boot/blob/v1.5.0.RC1/spring-boot-dependencies/pom.xml#L156) (at Ingalls-RC1, which is SD GemFire 1.9.0.RC1 and not SDG 2.0.0.M1, or Kay-M1 as required).

Other _Spring Session_ modules, like the samples applies the _[Spring Boot Gradle Plugin](https://github.com/spring-projects/spring-session/blob/master/samples/boot/build.gradle#L6-L8)_, which as the [documentation](http://docs.spring.io/spring-boot/docs/1.5.0.BUILD-SNAPSHOT/reference/htmlsingle/#build-tool-plugins-gradle-dependency-management) describes, also applies the Gradle [Dependency Management](https://github.com/spring-gradle-plugins/dependency-management-plugin/#dependency-management-plugin) plugin, thus enforcing the (LCD) versions of dependencies that _Spring Boot_ declares (at 1.5.0.RC1), thus overriding the SD BOM yet again.

Anyway, my goal was to not have to [specify](https://github.com/spring-projects/spring-session/blob/master/build.gradle#L23-L25) any _Spring Data_ module versions explicitly and let the BOM and (Gradle) dependency management handle it, but at the time, I could not figure it out properly.

As such, I will be opening another issue to set Spring Boot to 2.0.0, change Spring deps to build snapshots and correct the dependency management through proper application of the BOMs, as suggested in the (Gradle) [Dependency Management](https://github.com/spring-gradle-plugins/dependency-management-plugin#using-the-plugin) plugin management docs, as discussed with @rwinch.

3.  Finally, there were other issues in moving to SD Kay:

3.1. (SD) GemFire support required package renames from `com.gemstone.gemfire` to `org.apache.geode` on account of GemFire 9 (the new GemFire driver version in SDG Kay, or 2.0.0).  Additionally, I also resolved any issues around #672.

3.2. (SD) MongoDG support required significant `SessionRepository` changes since the underly MongoDG driver is not based on the [org.bson](http://api.mongodb.com/java/current/org/bson/package-summary.html) API (making `DBObject` and other types obsolete/deprecated).

3.3. Both (SD) Redis and MongoDB required Reactor on the classpath given the new Reactive extensions.

3.4. There were several other API breakage related fixes, but the above recaps the main ones.

This was a significant effort to overhaul, so I hope that we can keep the primary SD modules in-sync going forward.

Thanks!
